<?xml version="1.0"?>
<launch>
    <arg name="sim" default="false"/>
    <arg name="use_game_settings" default="false"/>
    <arg name="torquelessMode" default="false"/>
    <arg name="onlyImu" default="false"/>
    <arg name="onlyPressure" default="false"/>

    <rosparam file="$(find bitbots_ros_control)/config/wolfgang.yaml" command="load" />

    <rosparam param="torquelessMode" ns="/ros_control">$(arg torquelessMode)</rosparam>
    <rosparam param="onlyImu" ns="/ros_control">$(arg onlyImu)</rosparam>
    <rosparam param="onlyPressure" ns="/ros_control">$(arg onlyPressure)</rosparam>

    <node pkg="bitbots_ros_control" type="node" name="ros_control" output="screen"/>

    <rosparam file="$(find bitbots_ros_control)/config/controller.yaml" command="load" />
        <group if="$(arg use_game_settings)">
            <rosparam command="load" file="$(find bitbots_bringup)/config/game_settings.yaml" />
        </group>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
          output="screen" args="--timeout 20 joint_state_controller imu_sensor_controller DynamixelController ">
    </node>

    <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >    
        <rosparam command="load" file="$(find bitbots_ros_control)/config/analyzers.yaml" />
    </node>

    <include file="$(find bitbots_ros_control)/launch/imu.launch">
        <arg name="use_game_settings" value="$(arg use_game_settings)"/>
    </include>


    <node pkg="bitbots_ros_control" type="pressure_converter.py" name="pressure_converter" output="screen">
        <rosparam file="$(find bitbots_ros_control)/config/pressure_$(env ROBOT_NAME).yaml" command="load" />
    </node>
</launch>